frame/timestamp/src/lib.rs

```diff
@@ -167,35 +167,19 @@ decl_module! {
 			T::WeightInfo::set(),
 			DispatchClass::Mandatory
 		)]
-        fn set(origin, #[compact] now: T::Moment) {
-            ensure_none(origin)?;
-            assert!(!<Self as Store>::DidUpdate::exists(), "Timestamp must be updated only once in the block");
-            let prev = Self::now();
-
-            // information about block time is stored in inherents that are converted into
-            // extrinsics. Extrinsics from the first block are executed twice:
-            // - 1st time when there is no extrinsics in previous block (1st block)
-            // - 2nd time in second block when extrinsics from the previous block(1st block)
-
-            let timestamp = if prev.is_zero() {
-                // only happens for the first block - in order to have below assertion working
-                // lets modify timestamp of the first block so when it is executed for the second
-                // time assertion will be still correct
-                now - T::MinimumPeriod::get()
-            }else{
-                now
-            };
-
-            assert!(
-                prev.is_zero() || timestamp >= prev + T::MinimumPeriod::get(),
-                "Timestamp must increment by at least <MinimumPeriod> between sequential blocks"
-            );
-            <Self as Store>::Now::put(timestamp);
-            <Self as Store>::DidUpdate::put(true);
-
-            <T::OnTimestampSet as OnTimestampSet<_>>::on_timestamp_set(timestamp);
-        }
-
+		fn set(origin, #[compact] now: T::Moment) {
+			ensure_none(origin)?;
+			assert!(!<Self as Store>::DidUpdate::exists(), "Timestamp must be updated only once in the block");
+			let prev = Self::now();
+			assert!(
+				prev.is_zero() || now >= prev + T::MinimumPeriod::get(),
```

Comment:
If `let prev = Self::now();`, how can `now >= prev + T::MinimumPeriod::get()` be true?

Reply:
Actually i just reverted origin implementation(tags/v2.0.1) but i believe the first condition is for handling first block and the second one is for all the others